package.json

```diff
@@ -1,44 +1,63 @@
-    "eslint-plugin-json": "1.2.0",
-    "lintspaces-cli": "0.6.1",
-    "stylelint": "9.3.0",
-    "stylelint-config-standard": "18.2.0",
-    "webpack": "4.14.0",
-    "webpack-cli": "3.0.8",
-    "webpack-merge": "4.1.3"
-  },
-  "babel": {
-    "presets": [
-      [
-        "env",
-        {
-          "targets": {
-            "browsers": [
-              "ie >= 9"
-            ]
-          }
-        }
-      ]
-    ]
-  },
   "scripts": {
-    "build:dev": "webpack --config webpack-main-dev.config.js",
-    "build:prod": "webpack --config webpack-main-prod.config.js",
-    "build:test": "webpack --config webpack-test.config.js",
-    "dev": "npm run build:dev && npm run build:test",
-    "build": "npm run build:prod && npm run build:test",
-    "lint:js": "eslint *.js src/main/webapp/dev -c static-analysis/teammates-eslint.yml",
-    "lint:json": "eslint src/main/webapp/data/*.json src/main/resources/*.json src/test/resources/data/*.json -c static-analysis/teammates-eslint-json.yml",
-    "lint:css": "stylelint src/main/webapp/stylesheets/*.css --config static-analysis/teammates-stylelint.yml",
-    "lint:spaces": "lintspaces -n -t -d spaces -l 1 -. \"src/**/*.jsp\" \"src/**/*.tag\" \"src/main/**/*.html\" \"src/**/*.xml\" \"src/**/*.json\" \"src/**/*.css\" \"src/**/*.java\" \"src/main/webapp/dev/**/*.js\" \"src/**/*.properties\" \"*.js\" \"*.yml\" \"*.gradle\" \"static-analysis/*.*ml\"",
-    "lint": "npm run lint:js && npm run lint:json && npm run lint:css && npm run lint:spaces"
+    "ng": "ng",
+    "start": "ng serve",
+    "build": "ng build --prod && npm run assets",
+    "build:silent": "ng build --prod --progress=false && npm run assets",
+    "assets": "rm -rf src/main/webapp/assets && mv src/main/webapp/dist/assets src/main/webapp/assets",
+    "test": "ng test",
+    "coverage": "ng test --watch=false --code-coverage --browsers ChromeHeadless",
+    "lint:ts": "ng lint --tslint-config static-analysis/teammates-tslint.yml",
+    "lint:json": "jsonlint-cli src/web/**/*.json src/main/resources/*.json src/test/resources/data/*.json",
+    "lint:css": "stylelint \"src/web/**/*.scss\" --config static-analysis/teammates-stylelint.yml",
+    "lint:spaces": "lintspaces -n -t -d spaces -l 1 -. \"src/main/**/*.html\" \"src/web/**/*.html\" \"src/**/*.xml\" \"src/**/*.json\" \"src/**/*.properties\" \"*.yml\" \"*.gradle\" \"static-analysis/*.*ml\"",
+    "lint": "npm run lint:ts && npm run lint:json && npm run lint:css && npm run lint:spaces"
   },
-  "private": true
+  "private": true,
+  "dependencies": {
+    "@angular/animations": "^6.1.4",
+    "@angular/common": "^6.1.4",
+    "@angular/compiler": "^6.1.4",
+    "@angular/core": "^6.1.4",
+    "@angular/forms": "^6.1.4",
+    "@angular/http": "^6.1.4",
+    "@angular/platform-browser": "^6.1.4",
+    "@angular/platform-browser-dynamic": "^6.1.4",
+    "@angular/router": "^6.1.4",
+    "@ng-bootstrap/ng-bootstrap": "^3.1.0",
+    "bootstrap": "^4.1.3",
```

Comment:
Just realize we are in Bootstrp 4 🎉

Reply:
Well this is precisely why the UI is very disorganized right now. We'll need a serious work to get the UI back up to standard, but that's not my highest priority right now.